rust/ql/lib/codeql/rust/elements/internal/TypeParamImpl.qll

```diff
@@ -27,6 +28,36 @@ module Impl {
     /** Gets the position of this type parameter. */
     int getPosition() { this = any(GenericParamList l).getTypeParam(result) }
 
+    private int nrOfDirectTypeBounds() {
+      result = this.getTypeBoundList().getNumberOfBounds()
+      or
+      not this.hasTypeBoundList() and
+      result = 0
+    }
+
+    /**
+     * Gets the `index`th type bound of this type parameter, if any.
+     *
+     * This includes type bounds directly on this type parameter and bounds from
+     * any `where` clauses for this type parameter.
+     */
+    TypeBound getTypeBound(int index) {
+      exists(TypeBoundList tbl, int offset | result = tbl.getBound(index - offset) |
+        tbl = this.getTypeBoundList() and offset = 0
+        or
+        tbl = this.(TypeParamItemNode).getAWherePred().getTypeBoundList() and
+        offset = this.nrOfDirectTypeBounds()
```

Comment:
Again I think the index numbering breaks if there's more than one applicable `WherePred`.

Reply:
Should be fixed now :)